File: JS/Programmers/Level1/03 체육복.js
function solution(n, lost, reserve) {
  var answer = 0;

  let new_lost = new Array();
  let new_reserve = new Array();

  for (let i = 0; i < lost.length; i++) {
    if (reserve.indexOf(lost[i]) == -1) new_lost.push(lost[i]);
  }

  for (let i = 0; i < reserve.length; i++) {
    if (lost.indexOf(reserve[i]) == -1) new_reserve.push(reserve[i]);
  }

  new_lost.sort((a, b) => a - b);
  new_reserve.sort((a, b) => a - b);

  answer = n - new_lost.length;

  for (let i = 0; i < new_lost.length; i++) {
    if (
      new_reserve.includes(new_lost[i] - 1) == true &&
      new_lost.includes(new_lost[i] - 1) == false
    ) {
      answer++;
      new_reserve.splice(new_reserve.indexOf(new_lost[i] - 1), 1);
    } else if (
      new_reserve.includes(new_lost[i] + 1) == true &&
      new_lost.includes(new_lost[i] + 1) == false
    ) {
      answer++;
      new_reserve.splice(new_reserve.indexOf(new_lost[i] + 1), 1);
    }
  }

  return answer;
}

File: C++/Programmers/Level1/03 체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> new_lost, new_reserve;

    // 겹치는 번호가 없어야 새로운 배열에 추가
    for(int i=0; i<lost.size(); i++){
        if(find(reserve.begin(), reserve.end(), lost[i])==reserve.end()){
            new_lost.push_back(lost[i]);
        }
    }

    for(int i=0; i<reserve.size(); i++){
        if(find(lost.begin(), lost.end(), reserve[i])==lost.end()){
            new_reserve.push_back(reserve[i]);
        }
    }

    lost = new_lost;
    reserve = new_reserve;

    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());
    answer = n - lost.size();

    for(int i=0; i<lost.size(); i++){
        if((find(reserve.begin(), reserve.end(), lost[i]-1) != reserve.end()) && (find(lost.begin(), lost.end(), lost[i]-1) == lost.end())){
            answer++;
            reserve.erase(remove(reserve.begin(), reserve.end(), lost[i]-1), reserve.end());            
        }
        else if ((find(reserve.begin(), reserve.end(), lost[i]+1) != reserve.end()) && (find(lost.begin(), lost.end(), lost[i]+1) == lost.end())){
            answer++;
            reserve.erase(remove(reserve.begin(), reserve.end(), lost[i]+1), reserve.end());
        }
    }

    return answer;
}